### PR TITLE
migration from startActivityForResult to new API in ModelBrowser.java

### PR DESCRIPTION
**Pull request by divyansh-dxn**
## Purpose / Description
Remove dependency on deprecated code

## Fixes
Fixes a part of #8602

## How Has This Been Tested?
Passes unit tests and worked as expected on Realme 6i API 31

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

## Comparision before/after

* https://user-images.githubusercontent.com/69595691/158311235-e0240a88-8202-4a53-adc4-c528af54b642.mp4

* https://user-images.githubusercontent.com/69595691/158311219-46df391e-b854-4250-864d-50853f82c0e8.mp4

### CODE REVIEW

**Comment by divyansh-dxn** [comment]
In the file, on line 283, it uses another `startActivityForResult()` but the result was not handled inside `onActivityResult()`, neither the parent class handled its result code.

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.java [hunk 1]
```diff
@@ -499,7 +502,7 @@ private void dismissContextMenu() {
     private void openTemplateEditor() {
         Intent intent = new Intent(this, CardTemplateEditor.class);
         intent.putExtra("modelId", mCurrentID);
-        startActivityForResultWithAnimation(intent, REQUEST_TEMPLATE_EDIT, START);
+        launchActivityForResultWithAnimation(intent,mEditTemplateResultLauncher,START);
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.java:505]:
```suggestion
        launchActivityForResultWithAnimation(intent, mEditTemplateResultLauncher, START);
```

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.java [hunk 2]
```diff
@@ -611,11 +614,9 @@ public View getView(int position, @Nullable View convertView, @NonNull ViewGroup
         }
     }
 
-    @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        super.onActivityResult(requestCode, resultCode, data);
-        if (requestCode == REQUEST_TEMPLATE_EDIT) {
+    private final ActivityResultLauncher<Intent> mEditTemplateResultLauncher = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
+        if(result.getResultCode() == Activity.RESULT_OK) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.java:618]:
```suggestion
        if (result.getResultCode() == Activity.RESULT_OK) {
```

**Review by david-allison — CHANGES_REQUESTED** [review]
The PR needs to follow our Code Style: https://github.com/ankidroid/Anki-Android/wiki/Code-style

I've made two suggestions

Rest looks good

**Review by david-allison — APPROVED** [review]
Looks great, thanks!

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.java [hunk 3]
```diff
@@ -499,7 +502,7 @@ private void dismissContextMenu() {
     private void openTemplateEditor() {
         Intent intent = new Intent(this, CardTemplateEditor.class);
         intent.putExtra("modelId", mCurrentID);
-        startActivityForResultWithAnimation(intent, REQUEST_TEMPLATE_EDIT, START);
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.java:502]: is REQUEST_TEMPLATE_EDIT no unused? I think it is, and may be removed somewhere around line 60

**Review by mikehardy — CHANGES_REQUESTED** [review]
I think (could be wrong!) the constant definition may be deleted now?

> Passes unit tests and worked as expected on Realme 6i API 31

Thanks for filling in that question, that would have been my next question :+1:

**Review by mikehardy — APPROVED** [review]
Nice! If that compiles and passes CI it should be good to go!

**Pull request merged**